applications/CoSimulationApplication/python_scripts/solver_wrappers/external/acusolve_wrapper.py

```diff
@@ -0,0 +1,201 @@
+        inputFile           = self.settings["solver_wrapper_settings"]["input_file"].GetString()
+        np                  = self.settings["solver_wrapper_settings"]["num_processors"].GetInt()
+        nt                  = self.settings["solver_wrapper_settings"]["num_threads"].GetInt()
+        gpu                 = self.settings["solver_wrapper_settings"]["gpu_flag"].GetString()
+        rst                 = self.settings["solver_wrapper_settings"]["restart_flag"].GetString()
+        frst                = self.settings["solver_wrapper_settings"]["restart_flag"].GetString()
+        self.deltaT         = self.settings["solver_wrapper_settings"]["time_increment"].GetDouble()
+
+        # ---------------------
+        # Lauch AcuSolve
+        # ---------------------
+        if platform == "linux" or platform == "linux2":
+            if (gpu!="FALSE"):
+                cmd = "acuRun -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -gpu "+gpu+" -verbose "+str(echo_level)+" &"         
+            elif (rst!="FALSE"):
+                cmd = "acuRun -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -rst "+" -verbose "+str(echo_level)+" &"         
+            elif (frst!="FALSE"):
+                cmd = "acuRun -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -frst "+" -verbose "+str(echo_level)+" &"         
+            else:
+                cmd = "acuRun -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -verbose "+str(echo_level)+" &"       
+            cs_tools.cs_print_info(self.name + ": " +  cmd)
+            os.system(cmd)
+        elif platform == "win32":
+            if (gpu!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -gpu "+gpu+" -verbose "+str(echo_level)+" -lbuff"         
+            elif (rst!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -rst "+" -verbose "+str(echo_level)+" -lbuff"         
+            elif (frst!="FALSE"):
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -frst "+" -verbose "+str(echo_level)+" -lbuff"        
+            else:
+                cmd = "acuRun.bat -inp "+inputFile+" -pb "+problem+" -dir "+working_directory+" -pdir "+problem_directory+" -np "+str(np)+" -nt "+str(nt)+" -verbose "+str(echo_level)+" -lbuff"  
+            cs_tools.cs_print_info(self.name + ": " +  cmd)
+            subprocess.Popen(cmd)
```

Comment:
subprocess.run is the newer and safer alternative

Reply:
Okay